test/ServiceTest.cpp

```diff
@@ -0,0 +1,179 @@
+  // The code can be found in the history of this PR.
+  static auto constexpr getTsvFunctionFactory =
+      [](const std::string& expectedUrl, const std::string& expectedSparqlQuery,
+         const std::string& predefinedResult) -> Service::GetTsvFunction {
+    return [=](ad_utility::httpUtils::Url url,
+               const boost::beast::http::verb& method,
+               std::string_view postData, std::string_view contentTypeHeader,
+               std::string_view acceptHeader) -> std::istringstream {
+      // Check that the request parameters are as expected.
+      //
+      // NOTE: The first three are hard-coded in `Service::computeResult`, but
+      // the host and port of the endpoint are derived from the IRI, so the last
+      // two checks are non-trivial.
+      EXPECT_EQ(method, http::verb::post);
+      EXPECT_EQ(contentTypeHeader, "application/sparql-query");
+      EXPECT_EQ(acceptHeader, "text/tab-separated-values");
+      EXPECT_EQ(url.asString(), expectedUrl);
+
+      // Check that the (whitespace-normalized) post data contains the expected
+      // query.
+      //
+      // NOTE: a SERVICE clause specifies only the body of a SPARQL query, from
+      // which `Service::computeResult` has to construct a full SPARQL query by
+      // adding `SELECT ... WHERE`, so this checks something non-trivial.
+      EXPECT_EQ(
+          std::regex_replace(std::string{postData}, std::regex{"\\s+"}, " "),
+          expectedSparqlQuery);
+
+      return std::istringstream{predefinedResult};
+    };
+  };
+};
+
+// Test basic methods of class `Service`.
+TEST_F(ServiceTest, basicMethods) {
+  // Construct a parsed SERVICE clause by hand. The fourth argument is the query
+  // body (empty in this case because this test is not about evaluating a
+  // query). The fourth argument plays no role in our test (and isn't really
+  // used in `parsedQuery::Service` either).
+  parsedQuery::Service parsedServiceClause{{Variable{"?x"}, Variable{"?y"}},
+                                           Iri{"<http://localhorst/api>"},
+                                           "PREFIX doof: <http://doof.org>",
+                                           "{ }",
+                                           parsedQuery::GraphPattern{}};
+  // Create an operation from this.
+  Service serviceOp{testQec, parsedServiceClause};
+
+  // Test the basic methods.
+  ASSERT_EQ(serviceOp.getDescriptor(),
+            "Service with IRI <http://localhorst/api>");
+  ASSERT_EQ(serviceOp.getResultWidth(), 2);
+  ASSERT_EQ(serviceOp.getMultiplicity(0), 1);
+  ASSERT_EQ(serviceOp.getMultiplicity(1), 1);
+  ASSERT_EQ(serviceOp.getSizeEstimate(), 100'000);
+  ASSERT_EQ(serviceOp.getCostEstimate(), 1'000'000);
+  VariableToColumnMap varToColMap = serviceOp.computeVariableToColumnMap();
+  ASSERT_EQ(varToColMap.size(), 2);
+  ASSERT_EQ(varToColMap.at(Variable{"?x"}), 0);
+  ASSERT_EQ(varToColMap.at(Variable{"?y"}), 1);
```

Comment:
If you want to this can be something like
`ASSERT_THAT(varToColMap, ::testing::UnorderedElementsAre(....))`,
but this is probably a little bit difficult to set up.
Maybe we can write a custom matcher for all kinds of `VariableToColumnMaps`, because this kind of boilerplate appears in all unit tests. But I am not insisting on it:)